libkbfs/file_data.go

```diff
@@ -221,27 +221,28 @@ func (fd *fileData) getNextDirtyFileBlockAtOffset(ctx context.Context,
 	return ptr, parentBlocks, block, nextBlockStartOff, startOff, nil
 }
 
-// getLeafBlocksForOffsetRange fetches all the leaf ("direct") blocks
-// that encompass the given offset range (half-inclusive) in the file.
-// If `endOff` is -1, it returns blocks until reaching the end of the
-// file.  Note the range could be made up of holes, meaning that the
-// last byte of a direct block doesn't immediately precede the first
-// byte of the subsequent block.  If `prefixOk` is true, the function
-// will ignore context deadline errors and return whatever prefix of
-// the data it could fetch within the deadine.  Return params:
+// getBlocksForOffsetRange fetches all the blocks making up paths down
+// the file tree to leaf ("direct") blocks that encompass the given
+// offset range (half-inclusive) in the file.  If `endOff` is -1, it
+// returns blocks until reaching the end of the file.  Note the range
+// could be made up of holes, meaning that the last byte of a direct
+// block doesn't immediately precede the first byte of the subsequent
+// block.  If `prefixOk` is true, the function will ignore context
+// deadline errors and return whatever prefix of the data it could
+// fetch within the deadine.  Return params:
 //
 //   * pathsFromRoot is a slice, ordered by file offset, of paths from
 //     the root to each block that makes up the range.  If the path is
 //     empty, it indicates that pblock is a direct block and has no
 //     children.
 //   * blocks: a map from block pointer to a data-containing leaf node
-//     in the given range of offsets.
+//     in the given range of offsets, if `getDirect` is true.
 //   * nextBlockOff is the offset of the block that follows the last
 //     block given in `pathsFromRoot`.  If `pathsFromRoot` contains
 //     the last block among the children, nextBlockOff is -1.
-func (fd *fileData) getLeafBlocksForOffsetRange(ctx context.Context,
+func (fd *fileData) getBlocksForOffsetRange(ctx context.Context,
 	ptr BlockPointer, pblock *FileBlock, startOff, endOff int64,
-	prefixOk bool) (pathsFromRoot [][]parentBlockAndChildIndex,
+	prefixOk bool, getDirect bool) (pathsFromRoot [][]parentBlockAndChildIndex,
```

Comment:
in the `!pblock.IsInd` clause below, shouldn't you check `getDirect` for the returned `blocks` map?